arch/risc-v/src/bl602/bl602_spi.c

```diff
@@ -779,22 +785,37 @@ static void bl602_spi_dma_exchange(struct bl602_spi_priv_s *priv,
 static uint32_t bl602_spi_poll_send(struct bl602_spi_priv_s *priv,
                                     uint32_t wd)
 {
+  #warning Testing: BL602 SPI Timeout
   uint32_t val;
   uint32_t tmp_val = 0;
+  uint32_t timeout_cnt = SPI_RX_TIMEOUT_COUNT;
 
   /* write data to tx fifo */
 
   putreg32(wd, BL602_SPI_FIFO_WDATA);
 
-  while (0 == tmp_val)
+  /* wait for rx fifo ready or timeout */
+
+  while (0 == tmp_val && timeout_cnt > 0)
     {
-      /* get data from rx fifo */
+      /* get status of rx fifo */
 
       tmp_val = getreg32(BL602_SPI_FIFO_CFG_1);
       tmp_val = (tmp_val & SPI_FIFO_CFG_1_RX_CNT_MASK)
```

Comment:
You don't really need that shift, do you? That bit is either set or it's not; the value is of no consequence.

You could hoist the decrement and test of timeout_cnt into that loop itself. Then you could manage the entire life cycle (declaration, initialization, decrement, test, ensure no use-after-life) all in the loop body. Less spread out that way, but it's all style, I suppose. Some would find this uncomfortable.
for (uint32_t timeout_cnt = TIMEOUT_COUNT; !(getreg32(...CFG_1) & SPI_FIFO_CFG_1_RX_CNT_MASK) ;) {
  if (0 == timeout_cnt--) { info; return}
   
}



}

Reply:
Thanks for the note! I have scrapped this PR, the real reason why SPI Receive isn't working is because SPI Poll Send didn't configure the SPI Port correctly: https://github.com/lupyuen/incubator-nuttx/pull/42